src/App.jsx

```diff
@@ -14,45 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function sortGoods(goods, query) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (query) {
+      case SORT_FIELD_LENGTH:
+        return good1.length - good2.length;
+
+      case SORT_FIELD_NAME:
+        return good1.localeCompare(good2);
+
+      default:
+        return 0;
+    }
+  });
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortField);
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetButton = sortField || isReversed;
```

Comment:
```suggestion
  const isResetPossible = sortField || isReversed;
```
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/3d76c826-f4ab-4683-a2ad-77429c9b4715)